src/utils/create-broadcast.js

```diff
@@ -7,29 +7,36 @@
 
 export type Broadcast = {
   publish: (value: mixed) => void,
-  subscribe: (listener: (currentValue: mixed) => void) => () => void
+  subscribe: (listener: (currentValue: mixed) => void) => number,
+  unsubscribe: (number) => void,
 }
 
-const createBroadcast = (initialValue: mixed): Broadcast => {
-  let listeners = []
-  let currentValue = initialValue
+const createBroadcast = (initialState: mixed): Broadcast => {
+  const listeners = {}
+  let id = 0
+  let state = initialState
 
-  return {
-    publish(value: mixed) {
-      currentValue = value
-      listeners.forEach(listener => listener(currentValue))
-    },
-    subscribe(listener) {
-      listeners.push(listener)
+  const publish = (nextState: mixed) => {
+    state = nextState
+    const listenerKeys = Object.keys(listeners)
+    for (let i = 0; i < listenerKeys.length; i += 1) {
```

Comment:
Could you elaborate on this one? This seems to be the fastest approach via https://jsperf.com/object-keys-vs-for-in-with-closure/137. My one thought is it would be nice to not have to allocate the array of object keys, but on the other hand, publish'ing new themes seems like it wouldn't happen too often that it's a problem.